### DISCUSSION TITLE
how to suppress all warnings

### DISCUSSION THREAD

**Discussion by ChuyueSun** [dafny-lang/dafny#5037] [Q&A]
Is there a flag to turn on to avoid seeing all `dafny run` warnings?

**Comment by keyboardDrummer** [comment]
I don't think there is, but I understand your desire to have that. How would you feel if we send the error and warning output for `dafny run` to stderr instead of `stdout` ?

**Reply by ChuyueSun** [reply]
That would be great. Thanks!

**Reply by ChuyueSun** [reply]
BTW, is there a way to run dafny programs with "dafny run --no-verify" and not produce `json` and `dll`?

**Reply by keyboardDrummer** [reply]
> BTW, is there a way to run dafny programs with "dafny run --no-verify" and not produce json and dll?

Another good question ! I also don't think there's currently a way to do that. Dafny needs to place such intermediate files somewhere, but the current location, next to the source, is not great. I'm thinking of two alternatives, one is in a `build` folder, next to the sources, and the other is in the operating system's temporary files folder.

I'm inclined towards using a `build` folder, because it prevents any garbage from being placed in a folder that's hard to find for the user, and it theoretically allows Dafny to re-use artifacts from previous builds, in situations where the sources have changed so little that they can be reused.

What's your opinion?
